File: PSET1/cash/cash.c
#include <cs50.h>
#include <stdio.h>
#include <math.h>

#define QUARTER 25
#define DIME 10
#define NICKLE 5
#define PENNY 1

float get_change(string prompt);

int main(void)
{
    int coins = 0;
    
    float change = get_change("How much change: ");

    int rounded = round(change * 100);

    while (rounded / QUARTER > 0)
    {
        rounded = rounded - QUARTER;
        coins = coins + 1;
    }
    while (rounded / DIME > 0)
    {
        rounded = rounded - DIME;
        coins = coins + 1;
    }
    while (rounded / NICKLE > 0)
    {
        rounded = rounded - NICKLE;
        coins = coins + 1;
    }
    while (rounded / PENNY > 0)
    {
        rounded = rounded - PENNY;
        coins = coins + 1;
    }
    
    printf("%i\n", coins);
    
}

float get_change(string prompt)
{
    float f;
    do 
    {
        f = get_float("%s", prompt);
    }
    while (f < 0); 
    return f;
}

File: PSET6/cash/cash.py
from cs50 import get_float


def main():
    change = get_change("How much change: ")

    # Initialize coin count.
    coins = 0

    # Round change.
    rounded = (change * 100)

    # Check all types of coins.
    while rounded >= 25:
        rounded -= 25
        coins += 1

    while rounded >= 10:
        rounded -= 10
        coins += 1

    while rounded >= 5:
        rounded -= 5
        coins += 1

    while rounded >= 1:
        rounded -= 1
        coins += 1

    print(str(coins) + " coins")


# Prompt user change.
def get_change(prompt):
    while True:
        n = get_float(prompt)
        if n > 0:
            break
    return n


if __name__ == "__main__":
    main()